src/python/zquantum/core/wip/cost_function.py

```diff
@@ -154,9 +153,9 @@ def sum_expectation_values(expectation_values: ExpectationValues) -> ValueEstima
     precision = None
 
     if expectation_values.estimator_covariances:
-        estimator_variance = 0
+        estimator_variance = 0.0
         for frame_covariance in expectation_values.estimator_covariances:
-            estimator_variance += np.sum(frame_covariance, (0, 1))
+            estimator_variance += float(np.sum(frame_covariance, (0, 1)))
```

Comment:
I'm not sure how this line makes sense.
Why do we sum over `frame_covariance` and `(0,1)`?

Reply:
`frame_covariance` is a two-dimensional array where entry `i,j` is the covariance between term `i` and term `j`. To calculate the precision we have to sum over all pairs `i,j`, which corresponds to summing both over axis `0` and axis `1` of the array.